src/main/java/org/zeromq/ZMQ.java

```diff
@@ -1526,13 +1508,11 @@ public Poller(int size)
          */
         protected Poller(Context context, int size)
         {
-            if (context != null) {
-                this.context = context;
-            }
-
-            items = new PollItem[size];
-            timeout = -1L;
-            next = 0;
+            this.context = context;
```

Comment:
Add assert(context != null) would be nice checking and fit with the current code